app/src/main/java/org/mozilla/focus/utils/Browsers.java

```diff
@@ -71,8 +71,11 @@
     // installed then higher priority one will be stored here
     private ActivityInfo firefoxBrandedBrowser;
 
+    private Context context;
+
     public Browsers(Context context, @NonNull String url) {
         final PackageManager packageManager = context.getPackageManager();
+        this.context = context;
```

Comment:
Sorry, I missed this before. Please just pass the context as an argument to resolveBrowsers() instead of keeping a reference to it. The reason for that is that the Context we pass in is often an Activity object and this Browser object here might live longer than the Activity. If we keep a reference to the Activity here then we are keeping the (destroyed) Activity in memory.